commit 8e532175277d9a5eae49768ed086555081f741a7
Author: Jiri Slaby <jirislaby@gmail.com>
Date:   Mon May 4 18:04:55 2009 +0200

    mac80211: minstrel, fix memory corruption
    
    minstrel doesn't count max rate count in fact, since it doesn't use
    a loop variable `i' and hence allocs space only for bitrates found in
    the first band.
    
    Fix it by involving the `i' as an index so that it traverses all the
    bands now and finds the real max bitrate count.
    
    Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
    Cc: Felix Fietkau <nbd@openwrt.org>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/rc80211_minstrel.c b/net/mac80211/rc80211_minstrel.c
index 3824990..70df3dc 100644
--- a/net/mac80211/rc80211_minstrel.c
+++ b/net/mac80211/rc80211_minstrel.c
@@ -476,7 +476,7 @@ minstrel_alloc_sta(void *priv, struct ieee80211_sta *sta, gfp_t gfp)
 		return NULL;
 
 	for (i = 0; i < IEEE80211_NUM_BANDS; i++) {
-		sband = hw->wiphy->bands[hw->conf.channel->band];
+		sband = hw->wiphy->bands[i];
 		if (sband->n_bitrates > max_rates)
 			max_rates = sband->n_bitrates;
 	}
